sys/boot/geli/geliboot.c

```diff
@@ -334,6 +387,28 @@ geli_read(struct dsk *dskp, off_t offset, u_char *buf, size_t bytes)
 }
 
 int
+geli_havekey(struct dsk *dskp)
```

Comment:
this is used before the password prompt in the consumers of geliboot, to see if we need to prompt for the password or not